### PR TITLE
Add Chemistry Lab page(s)

### PR DESCRIPTION
**Pull request by Chew**
This is a draft PR, it's a big addition and it'll take me a while to do. Just wanted to get it up here to get your input as I make it.

Test Site: https://affectionate-kilby-ffc75c.netlify.com/chemistry-lab

Plans for this PR:

# New Pages
- [x] "Chemistry Lab" page, tells you about the lab and the equipment, and how to use them
- [x] Elements page
- [x] "Chemistry Lab Products", page of products you can make
- [x] "Chemistry Lab Compounds"

Potentially a place listing each block's reducer elements, as well as how many protons/neutrons/electrons you need for each element.

# Changed Pages
* Index File, add indexes for
  - [x] Chemistry Lab Page
  - [ ] Chemistry Craftables (?)
  - [x] balloon? (would link to above page, much like the 3-D Glasses)
  - [x] elements
  - [x] super fertiliser
  - [ ] glow sticks? (^)

### CODE REVIEW

**Comment by eccentricdevotion** [comment]
Looks good so far, only changes I'd make are:
- Vanilla resource pack for the GUI images
- Add a section about using TARDIS resource pack to see the items...

**Comment by Chew** [comment]
> * Vanilla resource pack for the GUI images

You know I end up forgetting I have the resource pack on lol

> * Add a section about using TARDIS resource pack to see the items...

I made a tiny note about it on the top, should I make it more pronounced?

**Comment by eccentricdevotion** [comment]
I think the TARDIS resource pack is more important for the GUI items than the room itself...

**Comment by Chew** [comment]
Okay, I finished the [chemistry lab page](https://affectionate-kilby-ffc75c.netlify.com/chemistry-lab) and started working on a [chemistry lab products page](https://affectionate-kilby-ffc75c.netlify.com/chemistry-lab-products).

I changed the GUI pictures to use the default textures. Anything else need to be changed on the Chemistry Lab page?

Check out what I've done for products, Lamps are done, and I plan on replicating that for the rest of the 7 products. Anything that needs to be changed before I do so?

**Comment by eccentricdevotion** [comment]
Just a couple of things you might not realise...
- Creative Chemistry GUI allows shift-clicking to add 64 items to player inventory
- Chemistry lamps can be toggled on and off with the Redstone upgraded Sonic Screwdriver :)

**Comment by Chew** [comment]
Alright, I've added sections for Balloon and Super fertiliser. Let me know what you think of those!

General questions:

1) What is the bleach used for?
2) How does the glow stick work?
3) What does the heat block do?
4) is the ice bomb just supposed to be a re-textured snowball?

Also, crafting bleach just shows a white dye and not the proper bleach texture

**Comment by eccentricdevotion** [comment]
Two things for new additions:
1. Spelling "pink" - missing a 'k'
2. If you hold a balloon in both hands then you get jump boost 2

Answers:
1. Making things white - same as education edition
2. Hold it, right click air, should change to glowing (but doesn't actually give off light)
3. Melts snow and ice
4. Ice bomb will freeze water when thrown into it
5. Odd, I crafted some earlier today and it had the correct bleach bottle model
bleach

**Comment by Chew** [comment]
Some things I've noticed in my testing

1) Putting items into any chemistry GUI wipes it on close. Kinda annoying when you expect them to either drop or go into inventory.
2) You can place pretty much any custom item. Ballons are flowers, so nothing is stopping you from placing them. 
3) "Bleaching" a leather tunic in a crafting table just applies white dye and doesn't reset it. That or I don't know how it works lol

**Comment by eccentricdevotion** [comment]
1. Yeah, should probably do something about that
2. I'd be hesitant to be checking _every_ placement on the server and checking whether it's a chemistry item, I can do a build for testing if you want try it out - maybe with a config option to load/not load at startup.
3. I haven't actually added recipes for bleaching leather armour, as the number of possible leather armour colours is pretty huge. Currently Beds, Wool, Carpet, Banners and Concrete Powder are supported.

File: elements.md [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+title: Atomic Elements
+---
+
+# Atomic Elements
+
+This is a cheat sheet for the Atomic Elements used in the [Element constructor](chemistry-lab#element-constructor)
+
+Please note, the Protons and Electrons are the same number and are based on the Atomic Number.
+
+Atomic Number = Protons = Electrons
```
Comment (eccentricdevotion) [inline elements.md:12]: Maybe just add an example to be super clear?
`e.g. Lithium has 3 protons, 4 neutrons and 3 electrons`

**Comment by Chew** [comment]
> Currently Beds, Wool, Carpet, Banners and Concrete Powder are supported.

Vanilla already supports normal white dye conversion from colored -> white already. That's why I tried the leather armor first

Responding to 2), almost every item doesn't seem to respect vanilla items. For example, the bio-scanner is a glowstone dust, so that + 3 vanilla glowstone dust still make a glowstone block. The sonic screwdriver can be converted into blaze powder, etc.

**Comment by Chew** [comment]
Also since we're adding, 6) The sparkler particles are a bit too far right

**Comment by Chew** [comment]
wew ok, did some more things

* added heat block and ice bomb sections to products page
* created and started working on the [chemistry lab compounds](https://affectionate-kilby-ffc75c.netlify.com/chemistry-lab-compounds) page. is the layout ok to continue?
* Added balloons, super fertiliser, chemistry lab (just main page) and elements to the site map / index files.

any updates to my questions/suggestions 4-8, i've got more

9) `/tchemistry gui formula Heat_Block` no longer shows Charcoal in the recipe, however it does need it to craft

File: chemistry-lab-products.md [hunk 2]
```diff
@@ -53,21 +51,70 @@ All recipes must be crafted in the [Product crafter](chemistry-lab#product-craft
 
 ## Glow Sticks
 
-Glow sticks
+Glow sticks are cosmetic items that don't really do much on their own. They don't provide light, however holding one and right clicking into the air will give it an enchanting effect. Unlike [Balloons](#balloons), there are only 14 colours.
+
+The following colours are available: White, Orange, Magenta, Light Blue, Yellow, Lime, Pink, Light Gray, Cyan, Purple, Blue, Brown, Green, and Red.
+
+![glow sticks](images/chemistry/products/glow-sticks.png)
+_All 14 colours of glow sticks. 2 are activates_
```
Comment (eccentricdevotion) [inline chemistry-lab-products.md:59]: activated

**Comment by eccentricdevotion** [comment]
> 1) Putting items into any chemistry GUI deletes the items when you close it

06a58e7 Fixed

> 2) You can place pretty much any custom item

005e7b3 Fixed, though due it being clientside you do see a flash before the server removes it

> 3) "Bleaching" a leather tunic in a crafting table just applies white dye and doesn't reset it

82a7106 Fixed, recipes for Leather Armour added, bleaching any now makes it just white

> 4) Clicking on a compound/product should lead to how to make it

Clicking where?

> 5) White Glowstick should have white dye in the center to match other recipes

2b436bc Fixed

> 6) The sparkler particles are a bit too far right

0a576d9 Better, but it's never going to be perfect

> 7) Using one sparkler consumes the whole stack

715384b Fixed

> 8) Super fertiliser in creative uses one item then doesn't use any more

60917e1 Fixed, it doesn't consume any in Creative (like vanilla bonemeal)

> 9) `/tchemistry gui formula Heat_Block` no longer shows Charcoal in the recipe

94a14ae Fixed

File: chemistry-lab-compounds.md [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+
+TIP: `/tchem` and `/tchemistry` are valid aliases of `/tardischemistry`
+
+## Resource Pack
+
+**_The compounds are best viewed with the [resource pack](https://github.com/eccentricdevotion/TARDIS-Resource-Pack)._**
+
+All images use the TARDIS-Resource-Pack, and your experience will be much better with it!
+
+## Table of Contents
```
Comment (eccentricdevotion) [inline chemistry-lab-compounds.md:22]: Can we use a table here and make the content two columns (like site map but without the ABCs)?

Comment (eccentricdevotion) [reply chemistry-lab-compounds.md:22]: Likewise for other TOCs

**Comment by eccentricdevotion** [comment]
100b96e
Chemistry GUI block recipes have always been there, but now you can view the recipe for them...

File: chemistry-lab-compounds.md (hunk 3 above)
Comment (Chew) [reply chemistry-lab-compounds.md:22]: oop, I sorted by letter anyway

**Comment by Chew** [comment]
image
All I did was try to place the creative atomic elements block a few times...

Also, table of contents much improved, let me know how it looks (only on compounds and products)

**Comment by Chew** [comment]
> 4. Clicking on a compound/product should lead to how to make it
Clicking when you're viewing a /tchemistry forumla. E.g. the one for bleach needs water, clicking water should lead you to how to make the water.

Introducing:

10. Bleach is missing the Sodium Hypochlorite from the recipe in /tchemistry formula Bleach

**Comment by eccentricdevotion** [comment]
Placing mushroom blocks (which is what the chemistry blocks are) next to each other will change the blockstate (clientside) the server switches it back again, but the client doesn't catch up - I couldn't get mine to go as extreme as your example - but right-clicking the affected blocks (or re-logging, or tp'ing to another world and back) updates the block texture/model...

**Comment by eccentricdevotion** [comment]
> 4. Clicking on a compound/product should lead to how to make it

5156ee3 Done 

> 10. 11. broken formulas

893f462 Fixed

**Comment by Chew** [comment]
Okay, Products page and Compounds page should be finished. May need to add some links.

Problem with Iron Sulfide, the name implies it's Iron and some amount of Sulfur (2 according to wikipedia), however the recipe doesn't work and `/tchemistry formula Iron_Sulfide` only shows 1 Iron.

**Comment by eccentricdevotion** [comment]
623b70e 1 iron 2 sulphur would be Iron Disulphide...

**Comment by Chew** [comment]
In my defense I am not a chemistry expert. Anything else you want me to do?

**Comment by eccentricdevotion** [comment]
No, it's looking great, thanks for all your work :)

**Comment by Chew** [comment]
AH i remembered, Charcoal reduces into 4 Nitrogen and 1 Oxygen or so, but requires 40 hydrogen to make? Shouldn't they be the same?

**Comment by Chew** [comment]
I also added quick Material reductions page. Slipped it right in. Shows all material reductions for the vanilla items and sorta. Useful!

File: material-reductions.md [hunk 4]
```diff
@@ -5,35 +5,37 @@ layout: default
 
 # Item/Block Material Reductions
 
+These are all vanilla items/blocks that can be reduced in the [material reducer](chemistry-lab#material-reducer). Useful for geting quick elements!
```
Comment (eccentricdevotion) [inline material-reductions.md:8]: getting

**Comment by eccentricdevotion** [comment]
> Charcoal reduces into 4 Nitrogen and 1 Oxygen or so, but requires 40 hydrogen to make

8f08768 Fixed

**Comment by eccentricdevotion** [comment]
Epic!

**Comment by Chew** [comment]
And finally, after 28 commits, like 30 messages, and numerous bugs squashed, this story comes to a close.

**Pull request merged**